substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/TimeZoneSubstitutions.java

```diff
@@ -78,12 +78,23 @@ public static TimeZone getTimeZone(String id) {
     static class Options {
         @Option(help = "When true, all time zones will be pre-initialized in the image.")//
         public static final HostedOptionKey<Boolean> IncludeAllTimeZones = new HostedOptionKey<>(false);
```

Comment:
Not sure this is the right default value. How much does `-H:+IncludeAllTimeZones` cost in terms of image size?

Anyhow, this is out of the scope of this PR.

Reply:
> How much does -H:+IncludeAllTimeZones cost in terms of image size?

We have been having this similar conversation in the Quarkus project https://groups.google.com/forum/#!msg/quarkus-dev/J1afzAhpShI/A5vmUGJLAQAJ

My experiments (with a couple of different programs, one being the `HelloWorld` example used in `substratevm` project) have shown that when `IncludeAllTimeZones` isn't set then the filesize is something along lines of 2.9M:

```
ls -lh helloworld 
-rwxr-xr-x  1 jaikiran  jaikiran   2.9M Nov 14 23:22 helloworld
```
and when you set `IncludeAllTimeZones`, the size increases very slightly to:
```
-rwxr-xr-x  1 jaikiran  jaikiran   3.3M Nov 14 23:50 ./helloworld
```

So not a big difference really. So yes, by default including all timezones probably will help application frameworks like Quarkus, where in certain situations there isn't really an easy way to determine at build time which timezones will be needed in the native image (imagine an application which communicates with a DB server whose timezone information can only be obtained at runtime).

But yes, like you say, the default value of `IncludeAllTimeZones` is something that needs to be discussed and sorted out separately.